packages/network-controller/src/NetworkController.ts

```diff
@@ -429,6 +455,77 @@ export class NetworkController extends BaseControllerV2<
     }
     return Promise.resolve(true);
   }
+
+  /**
+   * Network Configuration management functions
+   */
+
+  /**
+   * Adds a network configuration if the rpcUrl is not already present on an
+   * existing network configuration. Otherwise updates the entry with the matching rpcUrl.
+   *
+   * @param config - The network configuration.
+   * @param config.rpcUrl - The network configuration RPC URL.
+   * @param config.chainId - The chain ID of the network, as per EIP-155.
+   * @param config.ticker - Currency ticker.
+   * @param config.nickname - Personalized network name.
+   * @param config.rpcPrefs - Personalized preferences.
+   * @returns uuid for the added or updated network configuration
+   */
+  upsertNetworkConfiguration({
+    rpcUrl,
+    chainId,
+    ticker,
+    nickname,
+    rpcPrefs,
+  }: NetworkConfiguration): string {
+    const networkConfigurations = { ...this.state.networkConfigurations };
+
+    const newNetworkConfiguration: NetworkConfiguration = {
```

Comment:
TypeScript already "knows" that this object is a NetworkConfiguration because that's what's coming through in the arguments. Instead of assigning an explicit type here, what do you think about assigning `networkConfiguration: NetworkConfiguration` in the signature and then destructuring the properties here?

Reply:
I was guessing it was destructured to allow for proper JSDoc annotations. The linter complains when you try to describe object properties that aren't destructured. But if we don't annotate them here, they don't appear in the popover when writing this method in VSCode (or any other editor supporting JSDoc).

Another alternative would be removing the type here:

```suggestion
    const newNetworkConfiguration = {
```

Reply:
done [here](https://github.com/MetaMask/core/pull/1064/commits/ee8cfdcdd7743c9a3318ff52c845ea82101e5dd7)